lib/msf/core/session/provider/single_command_shell.rb

```diff
@@ -114,6 +114,16 @@ def set_is_echo_shell(timeout, command_separator)
     @is_echo_shell = res.include?(cmd)
   end
 
+  def shell_command_token_win32(cmd, timeout=10)
+    shell_command_token_base(cmd, timeout, '&')
+  end
+  module_function :shell_command_token_win32
```

Comment:
I'm confused here.  If the object was to present the same external methods for use, then `shell_command_token_win32` and `shell_command_token_unix` declarations should remain unchanged.  I'd assumed the goal was to effectively hide the new method `shell_command_token_base` to prevent it being called outside the module, enforcing the original methods as the public access.  My understanding was that effectively, marking a method as a `module_function` effectively made it private to a module, but after reading, it seems that `private` works in a module as well as a class, and `module_function` does something else.
That said, I'm not the ruby wizard of the team, and I may be missing something in the documentation I was able to find on `method_function` declarations.

Reply:
I may have misunderstood @smcintyre-r7 's request above about putting the functions under the `module_function` directive - can you clarify?

Reply:
After reading the documentation, it looks like maybe we are unclear on `module_function` ramifications.

The goal was to keep `shell_command_token_win32` and `shell_command_token_unix` as the public entry points to `shell_command_token_base` but hide `shell_command_token_base` so that only the module could call it.

We were thinking that the solution was to declare `shell_command_token_base` as a module function.
In reality, after reading some descriptions, we're not positive what declaring a method as a `method_function` would do to scope.

TL;DR: please keep the changes adding back `shell_command_token_win32` and  `shell_command_token_unix` but don't make anything a `module_function` until we know more about the deeper ramifications.